rules/legacy_xcodeproj.bzl

```diff
@@ -860,6 +860,7 @@ def _populate_xcodeproj_targets_and_schemes(ctx, targets, src_dot_dots, all_tran
         target_settings["BAZEL_LLDB_INIT_FILE"] = lldbinit_file
 
         if product_type == "application" or product_type == "app-extension":
+            target_settings["INFOPLIST_FILE"] = ""
```

Comment:
@justinseanmartin would you adding a comment about _why_ this is there? I think we know the other suggested fixes so LGTM